editor.planx.uk/src/pages/FlowEditor/lib/store/preview.ts

```diff
@@ -187,13 +188,15 @@ export const previewStore: StateCreator<
   canGoBack: (node: Store.node | null) => {
     // XXX: node is a required param until upcomingNodes().shift() is
     //      optimised/memoized, see related isFinalCard() comment below
-    const { flow, hasPaid, previousCard } = get();
-    return (
-      Boolean(node?.id) &&
-      flow[node!.id!]?.type !== TYPES.Confirmation &&
-      Boolean(previousCard(node)) &&
-      !hasPaid()
-    );
+    const { hasPaid, previousCard } = get();
+    return Boolean(node?.id) && Boolean(previousCard(node)) && !hasPaid();
```

Comment:
_Non-blocking Question:_

Is it worth removing the check that the component is a `Confirmation` from the `canGoBack`? Part of me wonders if although the `back` is hidden in this new approach the `canGoBack` should maybe still be able to be a source of truth over whether it should be possible?

This is maybe silly and over defensive so please feel free to ignore.

Reply:
Yeah it's a fair question - my first attempt did just rely on `canGoBack()`.

However there are instances where we want the empty back bar in place for layout reasons, but `canGoBack()` would return a `false` - for example the first component in a flow. Without this the layout "jumps" a little which ends up being pretty noticable.